File: src/dealloc.f90
!-----------------------------------------------------------------------
!
! MODULE: dealloc_module
!> @brief
!> This module controls the deallocation process of run-time arrays.
!
!-----------------------------------------------------------------------

MODULE dealloc_module

  USE global_module, ONLY: i_knd

  USE sn_module, ONLY: sn_deallocate

  USE data_module, ONLY: data_deallocate

  USE mms_module, ONLY: mms_deallocate

  USE geom_module, ONLY: geom_dealloc

  USE solvar_module, ONLY: solvar_dealloc

  USE control_module, ONLY: control_dealloc

  IMPLICIT NONE

  PUBLIC


  CONTAINS


  SUBROUTINE dealloc_input ( flg )

!-----------------------------------------------------------------------
!
! Call for the data deallocation from individual deallocation
! subroutines. Covers the allocations from input.
!
!-----------------------------------------------------------------------

    INTEGER(i_knd), INTENT(IN) :: flg
!_______________________________________________________________________

    CALL sn_deallocate

    IF ( flg > 1 ) CALL data_deallocate

    IF ( flg > 2 ) CALL mms_deallocate
!_______________________________________________________________________
!_______________________________________________________________________

  END SUBROUTINE dealloc_input


  SUBROUTINE dealloc_solve ( flg )

!-----------------------------------------------------------------------
!
! Call for the data deallocation from individual deallocation
! subroutines. Covers the allocations from input.
!
!-----------------------------------------------------------------------

    INTEGER(i_knd), INTENT(IN) :: flg
!_______________________________________________________________________

    CALL geom_dealloc

    IF ( flg > 1 ) CALL solvar_dealloc

    IF ( flg > 2 ) CALL control_dealloc
!_______________________________________________________________________
!_______________________________________________________________________

  END SUBROUTINE dealloc_solve


END MODULE dealloc_module

File: ports/snap-c/dealloc.c
/***********************************************************************
 * Controls the deallocation process of run-time arrays.
 ***********************************************************************/
#include "snap.h"

/***********************************************************************
 * Call for the data deallocation from individual deallocation
 * subroutines. Covers the allocations from input.
 ***********************************************************************/
void dealloc_input ( int selectFlag, sn_data *sn_vars, data_data *data_vars,
                     mms_data *mms_vars )
{
    sn_deallocate ( sn_vars );
    if ( selectFlag > 1 )
         data_deallocate ( data_vars );
    if ( selectFlag > 2 )
        mms_deallocate ( mms_vars );
}

/***********************************************************************
 * Call for the data deallocation from individual deallocation
 * subroutines. Covers the allocations from input.
 ***********************************************************************/
void dealloc_solve ( int selectFlag, geom_data *geom_vars,
                     solvar_data *solvar_vars, control_data *control_vars )
{
    geom_dealloc ( geom_vars );
    if ( selectFlag > 1 )
        solvar_dealloc ( solvar_vars );
    if ( selectFlag > 2 )
        control_dealloc ( control_vars );
}
